### ISSUE TITLE
404 not found for all dynamic route when using appDir.

### ISSUE THREAD

**Issue by nhunguyen15998** [aralroca/next-translate#1009]
When I add nextTranslate to the next config, the route alway return 404 not found. My route is /blog/[slug].js 

`const nextTranslate = require('next-translate-plugin')
module.exports = nextTranslate({ experimental: { appDir: true } })`

When I remove nextTranslate in next config everything work ok.

**Comment by aralroca** [comment]
Probably related with this: https://github.com/aralroca/next-translate/issues/1005

**Comment by aralroca** [comment]
Also (maybe) can be related to this Next.js issue: https://github.com/vercel/next.js/issues/46841

**Comment by middle0421** [comment]
Environment:
Node:v18.11.0
nextjs:8.19.2
next-translate-plugin@2.0.2
next-translate@2.0.2

From http://localhost:3000/en, change to http://localhost:3000/?lang=en work for me.
I am a beginner, so I just try some combinations.  I hope it will help solve the question.

**Comment by aralroca** [comment]
@middle0421 how is your middleware?

**Comment by middle0421** [comment]
@aralroca  I didn't change middleware.
First, I added the line in middleware, but it didn't work. Then I remove the line and tried the method that I wrote, and it worked.

**Comment by oknoorap** [comment]
The example directory seems doesn't works

**Comment by max-frai** [comment]
+ Same for me, doesn't work on latest nextjs

**Comment by smoglica** [comment]
Same. +1

**Comment by aralroca** [comment]
Try next-translate-plugin  [2.3.0-canary.1](https://github.com/aralroca/next-translate-plugin/releases/tag/2.3.0-canary.1) prerelease please 🙏

**Comment by aralroca** [comment]
I have made a PR to support `/[lang]` dynamic routing in an easy way.

You can try it with these versions:

- `next-translate` - version [**`2.3.0-canary.3`**](https://github.com/aralroca/next-translate/releases/tag/2.3.0-canary.3) 
- `next-translate-plugin` - version [**`2.3.0-canary.6`**](https://github.com/aralroca/next-translate-plugin/releases/tag/2.3.0-canary.6)

Things to keep in mind:

- The  pages defined inside `i18n.js` will have to be updated by adding the `/[lang]` in front of them.
- No middleware would be needed, unless you want to control a redirect.
- You have to put all current `/app` pages inside `/app/[lang]`.

I updated the example of next-translate working with `/app/[lang]` and Next.js 13.4:

https://github.com/aralroca/next-translate/blob/2078477ec299d1287cc20d5245e3781554a1ac39/examples/with-app-directory/

both ways now are supported (`?lang=` or `/[lang]`), but the way using `?lang=` needs middleware to do the rewrite and as of Next.js 13.3 it no longer works well, so I've added a second option for people to use until [the issue](https://github.com/vercel/next.js/issues/49883) is resolved by the Next.js team.

**Comment by jaisharx** [comment]
@aralroca I just tested the examples repo you have for the app dir: https://github.com/aralroca/next-translate/tree/master/examples/with-app-directory

Ran it with the script ```yarn example:with-app-directory``` in the main package.json file. But all I'm seeing is the default 404 page, which is exactly what I'm seeing in my personal projects where I'm using the package, which is what made me investigate it initially. It seems the issue was fixed a while back but now is somehow introduced again? Does your app router example works for you?

<img width="1652" alt="image" src="https://github.com/aralroca/next-translate/assets/55892116/7e893b42-cc2e-4beb-bbe7-b667ebcf051d">
